toast-tk-runtime/src/main/java/io/toast/tk/runtime/report/TemplateHelper.java

```diff
@@ -80,18 +111,74 @@ public static String getResultScreenshotAsBase64(final ITestResult testResult) {
 		return testResult.getScreenShot();
 	}
 
+	public static String getLineId(final TestLine line) {
+		return line.getId();
+	}
+
+	public static String formatSmallStringToHtml(final TestLine line) {
+		return substringText(formatStringToHtml(line), HTML_LENGTH_MAX);
+	}
+	
 	public static String formatStringToHtml(final TestLine line) {
 		if(line.getTestResult() != null) {
 			final String message = line.getTestResult().getMessage();
-			return message != null ? message.replace("\n", "<br>") : "";
+			return message != null ? prettyXmlText(message) : "";
 		}
 		return "&nbsp;";
 	}
 
+	public static String getSmallExpectedResult(final TestLine line) {
+		return substringText(getExpectedResult(line), HTML_LENGTH_MAX);
+	}
+	
+	public static String getExpectedResult(final TestLine line) {
+		final String message = line.getExpected(); 
+		return message != null ? prettyXmlText(message) : "";		
+	}
+	
+	public static String getSmallStepSentence(final TestLine line) {
+		return substringText(getStepSentence(line), 2*HTML_LENGTH_MAX);
+	}
+	
 	public static String getStepSentence(final TestLine line) {
 		final String contextualTestSentence = line.getTestResult() != null ? line.getTestResult().getContextualTestSentence() : null;
-		return contextualTestSentence == null ? line.getTest() : contextualTestSentence;
+		return contextualTestSentence != null ? contextualTestSentence : line.getTest();		
+	}
+
+	public static String substringText(String text, int maxLength) {
+		int textLength = text.length();
+		if(textLength < maxLength) {
+			return text;
+		}
+		else {
+			return text.substring(0, maxLength - 3) + "...";
+		}
 	}
+	
+	private static String prettyXmlText(String str) {
+		SAXBuilder builder = new SAXBuilder();
+		
+		//File fichierXML = new File("d:\\diagramme.xml");
+		InputStream stream;
+		try {
+			stream = new ByteArrayInputStream(str.getBytes("UTF-8"));
+			Document document;
+		
+			/* Parsing of the file */
+			//document = builder.build(fichierXML);
+			document = builder.build(stream);
+	
+			Element rootNode = document.getRootElement();
+
+			XMLOutputter output = new XMLOutputter();
+			output.setFormat(Format.getPrettyFormat());
+			String result = output.outputString(rootNode);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "result". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1488)